linode/manager
# upcoming: [M3-7459] - Disable Contact / Billing Info for Restricted Users

## Description 📝
Users with read-only access, should not have the ability to edit billing contacts or add payment methods. Currently, the "Edit" billing contact and "Add payment method" buttons are active for these types of users. While users receive denials via the API when trying to submit changes, this may creates confusion and frustration. To improve the user experience and align with intended permissions, it's necessary to disable these buttons.

## Changes  🔄
- `Edit` and `Add Payment Method` will be disabled for "Indirect Customers" and those with `read_only` account access.
- A tooltip will appear saying `To edit this section, please contact your {administrator || business partner}.`
- Fixed bug with disabled buttons not allowing us to tab out of them.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/125309814/d66f5287-dbb6-41f4-bc50-70ebfa4fe1cb" /> |![Screenshot 2024-01-05 at 1 41 56 PM](https://github.com/linode/manager/assets/125309814/926ec894-3523-4f2a-b332-03a73547d6dd) |
|Didn't want to update screenshot, but it's been updated to say `business partner`.| ![Screenshot 2024-01-05 at 1 43 23 PM](https://github.com/linode/manager/assets/125309814/12b07646-b6de-4ecc-bb81-0a6d55cdb9d4)|
||![Screenshot 2024-01-05 at 1 54 21 PM](https://github.com/linode/manager/assets/125309814/d8d7cdb9-01b8-4049-81af-3245ad9038a9)|

## How to test 🧪

### Prerequisites
- One account with full access
- One account with restricted access

### Reproduction steps
- Go to http://localhost:3000/account/billing

### Verification steps 
1. Log into any restricted account with parent/child feature flag **disabled**, observe the `Edit` and `Add Payment Method` buttons are disabled.
2. Turn parent/child feature flag on with Mock Service Workers
    - Observe the `Edit` and `Add Payment Method` buttons are enabled (by default we're in a `parent` state)
    - Then change `parent` to `child` on this line: https://github.com/linode/manager/blob/develop/packages/manager/src/mocks/serverHandlers.ts#L1236
    - Observe the `Edit` and `Add Payment Method` buttons are disabled.
    - Now change MSW from `child` to `proxy` and observe the `Edit` and `Add Payment Method` buttons are enabled.

## As an Author I have considered 🤔

*Check all that apply*
